proteus/Transport.py

```diff
@@ -5961,6 +5990,116 @@ def getMassResidual(self,u,r):
                 for dofN,g in self.dirichletConditionsForceDOF[cj].DOFBoundaryConditionsDict.iteritems():
                     r[self.offset[cj]+self.stride[cj]*dofN] = 0
 
+    def attachLaplaceOperator(self,nu=1.0):
+        """Attach a Discrete Laplace Operator to the Transport Class """
+
+        self.laplace_val = self.nzval.copy()
+        self.LaplaceOperator = SparseMat(self.nFreeVDOF_global,
+                                         self.nFreeVDOF_global,
+                                         self.nnz,
+                                         self.laplace_val,
+                                         self.colind,
+                                         self.rowptr)
+
+        _nd = self.coefficients.nd
+        if self.coefficients.nu != None:
+            _nu = self.coefficients.nu
+        self.LaplaceOperatorCoeff = DiscreteLaplaceOperator(nd=_nd)
+        _t = 1.0
+
+        Laplace_phi = {}
+        Laplace_dphi = {}
+        # ...initialize the phi functions...
+        for ci,space in self.testSpace.iteritems():
+            Laplace_phi[ci] = FiniteElementFunction(space)
+
+        for ck,phi in Laplace_phi.iteritems():
+            Laplace_dphi[(ck,ck)] = FiniteElementFunction(Laplace_phi[ck].femSpace)
+
+        for ci,dphi in Laplace_dphi.iteritems():
+            dphi.dof.fill(1.0)
+
+        self.Laplace_q = {}
+        scalar_quad = StorageSet(shape=(self.mesh.nElements_global,
+                                        self.nQuadraturePoints_element))
+        tensors_quad = StorageSet(shape={})
+        
+        scalar_quad |= set([('u',ci) for ci in range(self.nc)])
+        tensors_quad |= set([('a',ci,ci) for ci in range(self.nc)])
+        tensors_quad |= set([('da',ci,ci,ci) for ci in range(self.nc)])
+
+
+        scalar_quad.allocate(self.Laplace_q)
+        
+        for k in tensors_quad:
+            self.Laplace_q[k] = numpy.zeros(
+                (self.mesh.nElements_global,
+                 self.nQuadraturePoints_element,
+                 self.LaplaceOperatorCoeff.sdInfo[(k[1],k[2])][0][self.nSpace_global]),
+                'd')
+
+        if _nd == 2:
+            self.LaplaceOperatorCoeff.evaluate(_t,self.Laplace_q)
+        
+        LaplaceJacobian = {}
+        for ci in range(self.nc):
+            LaplaceJacobian[ci] = {}
+            for cj in range(self.nc):
+                if cj in self.LaplaceOperatorCoeff.stencil[ci]:
+                    LaplaceJacobian[ci][cj] = numpy.zeros(
+                        (self.mesh.nElements_global,
+                         self.nDOF_test_element[ci],
+                         self.nDOF_trial_element[cj]),
+                        'd')
+
+        for ci,ckDict in self.LaplaceOperatorCoeff.diffusion.iteritems():
```

Comment:
just for future discussion...this is going to have a constant null space since assembling only the element integrals is equivalent to pure homogenous Neumann conditions. It won't be that hard to grab dirichlet BC object for the pressure and set Dirichlet conditions along the proper boundary segment. I believe we're already expecting the constant null space when the Laplace operator appears in the PC, but I just wanted to flag this.